sql/parser/sql.y

```diff
@@ -1003,13 +1005,19 @@ set_reset_clause:
   SET set_rest {}
 
 show_stmt:
-  SHOW var_name
+  SHOW var_name on_privilege_target_clause for_grantee_clause
```

Comment:
fixed. the only reason putting `TABLES|DATABASES` explicitly caused reduce/reduce conflicts was the `SHOW TABLES FROM db` case which can combined with `SHOW TABLES` with the `FROM` clause optional.